### ISSUE TITLE
Specifying Appium Version when running an Appium test

### ISSUE THREAD

**Issue by guchen-doitt** [microsoft/appcenter#1594]
**Describe the solution you'd like**
App center should support multiple versions of Appium and should add a parameter to specify which Appium version I would like to use for the Appium test. The restriction of having Appium at exactly 1.11.0, Version 1.11.0 is from Feb 2019 -- one year old, does not contain all the bug fixes that Appium team has added since last year. Especially with a critical Android Espresso driver bug that was resolved with Appium 1.16

**Describe alternatives you've considered**
We can alter our scripts specifically for App Center so that out Android app runs on UIAutomator2 driver but this restricts how we test for attributes of the element under test. 

**Additional context**
The Appium version 1.11 that is being supported by app center is not compatible with some apps using the (android) espresso driver. The newest Appium 1.16 fixed a critical bug which allows espresso drivers to work properly for these apps again. The Espresso driver allows for a more robust unit / regression test for Android Apps because it can get attributes such as font color, font size, toasts and other attributes of elements that is not supported with the UIAutomator2 driver.

**Comment by Oddj0b** [comment]
Thank you for your feature request, is often you would want to run the later than the newest in your local setup?

**Comment by guchen-doitt** [comment]
Our organization has multiple automation units and each group is responsible for updating their own test suites. I would love to say that they always maintain scripts to the newest Appium version, but that is not the case.

**Comment by Oddj0b** [comment]
@guchen-doitt Thank you, and I understand the feature request better now.

**Comment by Zhappa** [comment]
> Our organization has multiple automation units and each group is responsible for updating their own test suites. I would love to say that they always maintain scripts to the newest Appium version, but that is not the case.

Please find the solution in this ticket: https://github.com/microsoft/appcenter/issues/1002
The support of Appium 1.16.0 added in experimental mode.

**Comment by guchen-doitt** [comment]
Great, thank you, will test this out.

**Comment by adriansuhov** [comment]
Hello! I see that now Appium is at 1.16.0 version in AppCenter by default. When running the test, either with '--test-parameter 'pipeline=appium-v1.16.0'' or without it, the test starts ok.
But when I try to use v.15.0 or something lower, i get this error:

> Starting test run... failed. 
> Error: Unknown pipeline branch: 'appium-v1.15.0'

I tried 1.15.0, 1.15.1 and 1.11.0. 

**Is the usage of older Appium version supported or not?**

**Comment by Oddj0b** [comment]
@adriansuhov We only support Appium version 1.16.0 https://docs.microsoft.com/en-gb/appcenter/test-cloud/appium/preparing-for-upload

**Comment by adriansuhov** [comment]
> 
> 
> @adriansuhov We only support Appium version 1.16.0 https://docs.microsoft.com/en-gb/appcenter/test-cloud/appium/preparing-for-upload

Thank you!

**Comment by Saranyanataraj** [comment]
Hi @Oddj0b, as per the latest update given in https://docs.microsoft.com/en-gb/appcenter/test-cloud/frameworks/appium/
Appium version 1.18.0 is only supported, but unfortunately this version of Appium is having issue while performing some actions on Android

**Bug reported on Appium side:** 
Android: longPress() TouchAction is not working (https://discuss.appium.io/t/long-press-expecting-a-int-when-passing-a-web-element/31632)
The above issue is fixed from Appium side in patch version v1.18.3. 

When can we expect AppCenter update for Appium, due to this issue most of my tests are failing in AppCenter. Please let us know asap. 

Thank you!

**Comment by Saranyanataraj** [comment]
> Hi @Oddj0b, as per the latest update given in https://docs.microsoft.com/en-gb/appcenter/test-cloud/frameworks/appium/
> Appium version 1.18.0 is only supported, but unfortunately this version of Appium is having issue while performing some actions on Android
> 
> **Bug reported on Appium side:**
> Android: longPress() TouchAction is not working (https://discuss.appium.io/t/long-press-expecting-a-int-when-passing-a-web-element/31632)
> The above issue is fixed from Appium side in patch version v1.18.3.
> 
> When can we expect AppCenter update for Appium, due to this issue most of my tests are failing in AppCenter. Please let us know asap.
> 
> Thank you!

@King-of-Spades could you please respond to us. I've been raising this concern with support team as well but haven't received any update.

**Comment by armartinez** [comment]
@Oddj0b I'm also having the same issue as @Saranyanataraj , any update on this ?

**Comment by King-of-Spades** [comment]
@Saranyanataraj Sorry for the delayed response, I think I saw the original @-mention but because of "auto-read" things in the inbox I forgot to respond. So for myself I'm actually changing teams, I still offer a little advice and encouragement to the support & engineering but am ramping up in a new role in GitHub, away from App Center.

I've just left a message for the App Center CSS team as well as our Test Engineering DRI linking directly to this discussion. In particularly I highlighted your observation that it appears an update to the Appium framework is required to mitigate the problem, though I haven't directly tested or investigated further personally.
